grc.fraud_web/src/main/java/com/typedb/examples/fraud/dao/BankDao.java

```diff
@@ -0,0 +1,67 @@
+package com.typedb.examples.fraud.dao;
+
+import com.typedb.examples.fraud.db.TypeDbSessionWrapper;
+import com.typedb.examples.fraud.model.Bank;
+import com.typedb.examples.fraud.model.BankCoordinates;
+import java.util.Hashtable;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+
+@RequestScoped
+public class BankDao {
```

Comment:
Agreed. An interface would enforce consistency between our DAO classes. We might be able to take it further, creating an `AbstractDao` abstract class with generic implementations of the aforementioned methods. This would allow us to reduce code duplication and keep implementations of the same method aligned.

Reply:
Agree. We'll add an interface.

Reply:
FYI, I'm adding an interface, but just for `getAll` and `insertAll`.

- `fromResult` is protected as it is intended to be used only by other Daos and statically (as we need to avoid instantiating the Daos ourselves).
- `getInsertQueryStr` is private as it should only be used internally, so not part of the public interface.